server/src/Config.hs

```diff
@@ -1,46 +1,72 @@
 module Config (
   Config (..),
-  configInfo
+  MetricsConfig (..),
+  configInfo,
+  metricsServerConfig
 ) where
 
 import Control.Applicative (optional)
-import Data.Semigroup      ((<>))
-import Data.List           (lookup)
-import Options.Applicative (Parser, ParserInfo, fullDesc, header, help,
-  helper, info, long, switch, metavar, strOption, value)
+import Data.Function ((&))
+import Data.Semigroup ((<>))
+import Options.Applicative
+import qualified Network.Wai.Handler.Warp as Warp
+import qualified Text.Read as Read
+import Data.String (fromString)
+import qualified Data.List as List
 import qualified Data.Text as Text
-
 import qualified Web.JWT as JWT
 
 -- command-line arguments
-data Config = Config {
-    configDataFile :: FilePath,
+data Config = Config
+  { configDataFile :: FilePath
     -- | Enables the use of JWT for authorization in JWT.
-    configEnableJwtAuth :: Bool,
+  , configEnableJwtAuth :: Bool
     -- | The secret used for verifying the JWT signatures. If no secret is
     -- specified even though JWT authorization is enabled, tokens will still be
     -- used, but not be verified.
-    configJwtSecret :: Maybe JWT.Secret
-}
+  , configJwtSecret :: Maybe JWT.Secret
+  , configMetricsEndpoint :: Maybe MetricsConfig
+  }
+
+data MetricsConfig = MetricsConfig
+  { metricsConfigHost :: Warp.HostPreference
+  , metricsConfigPort :: Warp.Port
+  }
+
+metricsServerConfig :: MetricsConfig -> Warp.Settings
+metricsServerConfig cfg = Warp.defaultSettings
+  & Warp.setHost (metricsConfigHost cfg)
+  & Warp.setPort (metricsConfigPort cfg)
+
+metricsConfigReader :: ReadM MetricsConfig
+metricsConfigReader = eitherReader $ \input ->
+  case List.findIndex (== ':') input of
+    Nothing -> Left "no port specified"
+    Just idx -> let (hostStr, portStr) = List.splitAt idx input
+                in MetricsConfig (fromString hostStr) <$> Read.readEither (drop 1 portStr)
```

Comment:
Could also be:

``` haskell
metricsConfigReader = eitherReader $ \input ->
  let (hostStr, portStr) = Data.List.break (== ':') input in  
  case List.findIndex (== ':') input of
     (_, "") -> Left "no port specified"
     Just (hostStr, portStr) -> MetricsConfig (fromString hostStr) <$> Read.readEither (drop 1 portStr)
```

Reply:
Using `break` is a good idea, although there's something wrong with your code snippet.